packages/adblocker/src/resources.ts

```diff
@@ -62,57 +84,88 @@ export default class Resources {
     });
   }
 
-  public static parse(data: string, { checksum }: { checksum: string }): Resources {
-    const typeToResource: Map<string, Map<string, string>> = new Map();
-    const trimComments = (str: string) => str.replace(/^\s*#.*$/gm, '');
-    const chunks = data.split('\n\n');
-
-    for (const chunk of chunks) {
-      const resource = trimComments(chunk).trim();
-      if (resource.length !== 0) {
-        const firstNewLine = resource.indexOf('\n');
-        const split = resource.slice(0, firstNewLine).split(/\s+/);
-        const name = split[0];
-        const type = split[1];
-        const body = resource.slice(firstNewLine + 1);
-
-        if (name === undefined || type === undefined || body === undefined) {
-          continue;
+  public static resolveScriptlet(scriptlet: Scriptlet, scriptlets: Scriptlet[]) {
```

Comment:
Understanding question. The problem is given a scriplet and a list of all scriptlets, to find the dependencies and resolve them in the correct order? In other words, a topological sorting (https://en.wikipedia.org/wiki/Topological_sorting)?

The function mentions circular relations at one point. Can scriptlets have cyclical dependencies?